src/main/java/httpserver/router/Router.java

```diff
@@ -0,0 +1,37 @@
+package httpserver.router;
+
+import httpserver.utils.Methods;
+import httpserver.models.Request;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Router {
+    Map<String, String[]> routes;
+
+    public Router() {
+        this.routes = addRoutes();
+    }
+
+    public Map<String, String[]> addRoutes() {
+        Map<String, String[]> routes = new HashMap<>();
+
+        routes.put("/simple_get", new String[]{Methods.GET, Methods.HEAD});
+        routes.put("/simple_get_with_body", new String[]{Methods.GET});
+        return routes;
+    }
+
+    public boolean isRequestValid(Request request) {
+        for (Map.Entry<String, String[]> set : routes.entrySet()) {
+            if (set.getKey().equals(request.path)) {
+                String[] methodsForPath = set.getValue();
+                for (int i = 0; i < methodsForPath.length; i++) {
+                    if (methodsForPath[i].equals(request.method)) {
+                        return true;
+                    }
```

Comment:
This is a nested for loop and has exponential time complexity. Is there a faster way to do this?

Reply:
Ah yes, I had to switch the HashMap from taking two strings to taking a string and an array, which is what I'm iterating over. Let's discuss in our meeting today.